util/dnsname/dnsname.go

```diff
@@ -65,6 +68,30 @@ func ToFQDN(s string) (FQDN, error) {
 	return FQDN(raw), nil
 }
 
+// ToFQDN strips a leading '.' on a string before converting to a FQDN.
+func ToFQDN(s string) (FQDN, error) {
+	if len(s) == 0 || s == "." {
+		return FQDN("."), nil
```

Comment:
The real problem is the FQDN type has no docs. But then these docs read contradictorily: it says it strips a leading dot but then these two paths return it with a leading dot. But if we do what an "FQDN" was (the undocumented type) then perhaps it'd make sense.